pkg/distributor/distributor.go

```diff
@@ -270,10 +274,18 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 	subservices := []services.Service(nil)
 	subservices = append(subservices, haTracker)
 
+	var requestBufferPool util.Pool
+	if cfg.MaxRequestPoolBufferSize > 0 {
+		requestBufferPool = util.NewBucketedBufferPool(1e3, cfg.MaxRequestPoolBufferSize, 2)
```

Comment:
I assume the third param is the factor here. Should it also be configurable?

I find `2` very aggressive: do we really want separate pools for 1000 and 2000 bytes slices? Such granularity can result being counter-productive as there will be more pools with less chances to reuse the slices.

Reply:
In that case, I might opt to be a bit less aggresive and increase the factor to `4`. 

But personally, I would prefer not to add it as a new config option, as I'd like to focus first on finding a default maximum that can accommodate most scenarios and, above all, prevent any OOM condition.

Later, we could refine this factor as a separate experiment. WDYT?

Reply:
_changed @ 43e8ece1a0f71f34ef57cba29ea263d5e4cb7bd5_